app/scripts/services/applicationGenerator.js

```diff
@@ -402,6 +402,55 @@ angular.module("openshiftConsole")
       return service;
     };
 
+  // Public method for creating a persistent volume claim.
+    // Expects pvcOptions from the osc-persistent volume claim directive.
+    scope.createPersistentVolumeClaim = function(pvcOptions, serviceName, labels) {
+      return scope._generatePersistentVolumeClaim({
+        labels: labels || {},
+        params: angular.extend({ include: true }, pvcOptions.persistentVolumeClaimForm)
+      }, pvcOptions.persistentVolumeClaimForm.name, serviceName);
+    };
+
+    scope._generatePersistentVolumeClaim = function(input, name, serviceName){
+      if(!input.params.include) {
+        return null;
+      }
+
+      var pvc = {
+        kind: "PersistentVolumeClaim",
+        apiVersion: "v1",
+        metadata: {
+          name: name.$viewValue,
```

Comment:
If you have `ng-model` on the input, you should be able to access it as a scope variable like `$scope.claim.name`.